driver/src/main/scala/api/indexes.scala

```diff
@@ -122,26 +124,18 @@ case class Index(
  * @param namespace The fully qualified name of the indexed collection.
  * @param index The other fields of the index.
  */
-case class NSIndex(
-    namespace: String,
-    index: Index) {
+case class NSIndex(namespace: String, index: Index) {
   val (dbName: String, collectionName: String) = {
     val spanned = namespace.span(_ != '.')
     spanned._1 -> spanned._2.drop(1)
   }
 }
 
-/**
- * A helper class to manage the indexes on the given database.
- *
- * @param db The subject database.
- */
-class IndexesManager(db: DB)(implicit context: ExecutionContext) {
-  val collection = db("system.indexes")
+/** Indexes manager at database level. */
+sealed trait IndexesManager {
```

Comment:
Retrofit it as trait, to have different impl (currently legacy < MongoDB 3, default >= MongoDB 3)